docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+
+An example syntax is:
+```yaml
+- matchCapabilities:
+  - type: Effective
+    operator: In
+    values:
+    - "CAP_CHOWN"
+    - "CAP_NET_RAW"
+```
+
+This will match if: [`Effective` capabilities contain `CAP_CHOWN`] OR
+[`Effective` capabilities contain `CAP_NET_RAW`]
+
+- `type` can be: `Effective`, `Inheritable`, or `Permitted`.
+- `operator` can be `In` or `NotIn`
+- `values` can be any supported capability. A list of all supported
+  capabilities can be found in `/usr/include/linux/capability.h`.
+
+**Limitations**
+
+1. There is no limit in the number of capabilities listed under `values`.
+2. Only one `type` field can be specified under `matchCapabilities`.
+
+#### Namespace changes filter
+
+Namespace changes filter can be specified under the `matchNamespaceChanges`
+field and provide filtering based on calls that are changing Linux namespaces.
+This filter can be especially useful to track container escape or execution of
+code in a new namespace, for example using user namespace to exploit root-only
+kernel features.
+
+```yaml
+matchNamespaceChanges:
+- operator: In
+  values:
+  - "Mnt"
+```
+
+The `unshare` command, or executing in the host namespace using `nsenter` can
+be used to test this feature. See a
+[demonstration example](https://github.com/cilium/tetragon/blob/c511bf632ee58d401aa0f0cf1df50b33760515e3/crds/examples/match_namespace_changes.yaml)
+of this feature.
+
+#### Capability changes filter
+
+Capability changes filter can be specified under the `matchCapabilityChanges`
+field and provide filtering based on calls that are changing Linux capabilities.
+This filter can be especially useful to detect privilege escalation.
+
+```yaml
+matchCapabilityChanges:
+- type: Effective
+  operator: In
+  isNamespaceCapability: false
+  values:
+  - "CAP_SETUID"
+```
+
+See a [demonstration
+example](https://github.com/cilium/tetragon/blob/c511bf632ee58d401aa0f0cf1df50b33760515e3/crds/examples/match_capability_changes.yaml)
+of this feature.
+
+#### Actions filter
+
+Actions filters are a list of actions that execute when an appropriate selector
+matches. They are defined under `matchActions` and currently, the following
+`action` types are supported:
+- [Sigkill action](#sigkill-action)
+- [Override action](#override-action)
+- [FollowFD action](#followfd-action)
+- [UnfollowFD action](#unfollowfd-action)
+- [CopyFD action](#copyfd-action)
+- [GetUrl action](#geturl-action)
+- [DnsLookup action](#dnslookup-action)
+- [Post action](#post-action)
+
+Note that `Sigkill`, `Override`, `FollowFD`, `UnfollowFD`, `CopyFD` and `Post`
+are executed directly in the kernel BPF code while `GetUrl` and `DnsLookup` are
+happening in userspace after the reception of events.
+
+##### Sigkill action
+
+`Sigkill` action terminates synchronously the process that made the call that
+matches the appropriate selectors from the kernel. In the example below, every
+`__x64_sys_write` system call with a PID not equal to 1 or 0 attempting to
```

Comment:
I guess these are the processes inside a container - so not host and not init and its children.

Reply:
here we have the PID 0 thing again... interesting! I wonder why.

comment added on `kubectl exec`!